WireAPI/Sources/WireAPI/APIs/SelfUserAPI/SelfUserAPIV0.swift

```
@@ -0,0 +1,159 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class SelfUserAPIV0: SelfUserAPI, VersionedAPI {
+
+    let httpClient: HTTPClient
+
+    init(httpClient: HTTPClient) {
+        self.httpClient = httpClient
+    }
+
+    var apiVersion: APIVersion {
+        .v0
+    }
+    
+    func getSelfUser() async throws -> SelfUser {
+        let request = HTTPRequest(
+            path: "\(pathPrefix)/self",
+            method: .get
+        )
+        
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: 200, type: SelfUserV0.self)
+            .failure(code: 404, label: "not-found", error: SelfUserAPIError.selfUserNotFound)
+            .parse(response)
+        
+    }
+    
+    func pushSupportedProtocols(_ supportedProtocols: Set<SupportedProtocol>) async throws {
+        guard apiVersion >= .v5 else {
+            throw SelfUserAPIError.unsupportedEndpointForAPIVersion
+        }
+        
+        let encoder = JSONEncoder.defaultEncoder
+        let payload = SupportedProtocolsPayloadV5(supportedProtocols: supportedProtocols)
+        let body = try encoder.encode(payload)
```

This implementation is for v5 so I suggest to move this to SelfUserAPIV5. Instead in this method we could simply throw the error without checking the api version.

---

makes sense thanks 👍

---

done